File: examples/sensor.py
import random, time
while True:
    # Wait 0 to 5 seconds
    time.sleep(random.random() * 5)
    # Set any temperature from -5 to 15
    temperature = (random.random() * 20) -5
    # We flush in order to ensure that the information arrives when we expect it to.
    print(temperature, flush=True, end='')

File: examples/sensor.js
function reportReading() {
    // Pick any temperature from a range of -5 to 15
    const temperature = (Math.random() * 20) - 5;
    // Write with newline char
    process.stdout.write(temperature + '\n');
    // Wait 0 to 5 seconds
    setTimeout(reportReading, Math.random() * 5000);
}
reportReading();
